hare/hare_test.go

```diff
@@ -414,6 +418,14 @@ func TestHare_onTick_notMining(t *testing.T) {
 	out := <-h.blockGenCh
 	require.Equal(t, lyrID, out.Layer)
 	require.ElementsMatch(t, types.ToProposalIDs(pList), out.Proposals)
+	require.Eventually(t, func() bool {
+		select {
+		case <-wcSaved:
+			return true
+		default:
+			return false
+		}
+	}, 5*time.Second, 10*time.Millisecond)
```

Comment:
```suggestion
	select {
	case <-wcSaved:
	case <-time.After(10*time.Second):
		require.Fail(t, "Weakcoin::Set has unexpectedly not been called")
	}
```

`require.Eventually` executes this function every 10 ms. This select has the same effect, but doesn't cause the runtime to be busy calling the function.